plugin/src/test/java/org/opensearch/ml/processor/MLInferenceIngestProcessorTests.java

```diff
@@ -138,6 +139,60 @@ public void testExecute_Exception() throws Exception {
 
     }
 
+    /**
+     * Models that use the parameters field need to have a valid NamedXContentRegistry object to create valid MLInputs. For example
+     * <pre>
+     * PUT   /_plugins/_ml/_predict/text_embedding/model_id
+     *  {
+     *     "parameters": {
+     *         "content_type" : "query"
+     *     },
+     *     "text_docs" : ["what day is it today?"],
+     *     "target_response" : ["sentence_embedding"]
+     *   }
+     * </pre>
+     * These types of models like Local Asymmetric embedding models use the parameters field.
+     * And as such we need to test that having the contentRegistry throws an exception as it can not
+     * properly create a valid MLInput to perform prediction
+     *
+     * @implNote If you check the stack trace of the test you will see it tells you that it's a direct consequence of xContentRegistry being null
+     */
+    public void testExecute_xContentRegistryNullWithLocalModel_throwsException() throws Exception {
```

Comment:
Can't we add another test for success case with Asymmetric model?

Reply:
Yes its possible but that would involve hosting the model zip somewhere so we can test that it makes embeddings.

Reply:
Can we take some idea from here: https://github.com/opensearch-project/ml-commons/pull/2123/files#diff-cd8365f3263802b111604d62b751e5264a2e51df44b46b9fb650ca001466328a

Reply:
Oh! I didn't realize the community user did that, I can definitely use that thanks!